spec/services/facility_stats_service_spec.rb

```diff
@@ -39,81 +22,136 @@ def facilities_data(facilities)
   describe "self.call" do
     it "sets default values for facilities_data and stats_by_size when no facilities are provided" do
       stats_by_size = FacilityStatsService.call(facilities: {}, period: period,
-                                                rate_numerator: "controlled_patients")
+                                                rate_numerator: :controlled_patients)
       expect(stats_by_size).to eq({})
     end
 
     it "sets data for the past six periods" do
       stats_by_size = FacilityStatsService.call(facilities: facilities_data([small_facility]), period: period,
-                                                rate_numerator: "controlled_patients")
-      small = stats_by_size["small"]
+                                                rate_numerator: :controlled_patients)
+      small = stats_by_size[:small][:periods]
       periods = (1..5).inject([period]) { |periods|
         periods << periods.last.previous
       }
       expect(small.keys).to match_array(periods)
     end
 
     it "accurately tallies stats for facilities by size and period" do
-      small_facilities = setup_for_size("small", december - 5.months)
-      medium_facilities = setup_for_size("medium", december - 4.months)
-      large_facilities = setup_for_size("large", december - 3.months)
+      small_facility1 = create(:facility, name: "small_1", facility_size: "small")
```

Comment:
In addition to moving the data setup out of the helper method, I've also added some more variety of data (i.e., different numbers of patients for each facility size).